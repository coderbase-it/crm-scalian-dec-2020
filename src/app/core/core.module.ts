import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { LoginModule } from '../login/login.module';
import { UiModule } from '../ui/ui.module';
import { IconsModule } from '../icons/icons.module';
import { TextsModule } from '../texts/texts.module';
import { HeaderComponent } from './components/header/header.component';
import { NavComponent } from './components/nav/nav.component';
import { FooterComponent } from './components/footer/footer.component';
import { RouterModule } from '@angular/router';



@NgModule({
  declarations: [HeaderComponent, NavComponent, FooterComponent],
  imports: [
    CommonModule,
    RouterModule
  ],
  exports: [
   // LoginModule,
    UiModule,
    IconsModule,
    TextsModule,
    HeaderComponent,
    NavComponent,
    FooterComponent
  ]
})
export class CoreModule { }
