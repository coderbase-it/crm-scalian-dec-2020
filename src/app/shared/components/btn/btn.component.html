<a *ngIf="label" [routerLink]="route">{{ label }}</a>
