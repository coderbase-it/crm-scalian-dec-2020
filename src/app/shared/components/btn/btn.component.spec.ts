import { ComponentFixture, TestBed } from '@angular/core/testing';
import { By } from '@angular/platform-browser';
import { RouterTestingModule } from '@angular/router/testing';

import { BtnComponent } from './btn.component';

describe('BtnComponent', () => {
  let component: BtnComponent;
  let fixture: ComponentFixture<BtnComponent>;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports:[RouterTestingModule],
      declarations: [ BtnComponent ]
    })
    .compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(BtnComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should test if tag a is generate when label exist', () => {
    component.label='label';
    component.route='route';
    fixture.detectChanges();
    const lien=fixture.debugElement.query(By.css('a'));
    // console.log('lien =>',lien.nativeElement.getAttribute('href'));
    expect(lien).toBeTruthy();
    expect(lien.nativeElement.getAttribute('href')).toEqual('/route');
  });

  it('should done snapshot', () => {
    component.label='label';
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });

});
