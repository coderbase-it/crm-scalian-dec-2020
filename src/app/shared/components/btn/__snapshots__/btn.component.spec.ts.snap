// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BtnComponent should done snapshot 1`] = `
<app-btn
  label={[Function String]}
>
  <a
    href="/"
  >
    label
  </a>
</app-btn>
`;
